File: week-04/java/day-1/src/DivideByZero.java
// Create a function that takes a number
// divides ten with it,
// and prints the result.
// It should print "fail" if the parameter is 0

public class DivideByZero {
    public static void main(String[] args) {
        divideByZero(2);
    }

    public static void divideByZero(int num) {
        try{
            int result = 10 / num;
            System.out.println(result);
        } catch(ArithmeticException e){
            System.out.println(e.getMessage());
        }
    }
}

File: week-04/typescript/day-1/divide-by-zero.ts
// Create a function that takes a number
// divides ten with it,
// and prints the result.
// It should print 'fail' if the parameter is 0

function divideByZero(num: number): void{
  try {
    if (num === 0){
      throw new Error("fail")
    }
    let result: number = 10 / num;
    console.log(result);
  } catch(e){
    console.log(e.message);
  }
}

divideByZero(0);
divideByZero(2);